### PR TITLE
Basis vector colours

### PR DESCRIPTION
**Pull request by dagewa**
While poking around inside the image viewer for another branch, I noticed that when there are multiple experiments, the crystal basis vectors were always shown with one colour, whereas the predictions cycle through a set of colours. This PR changes the basis vector display to match the prediction colours.

**Before**
main

**After**
branch

### CODE REVIEW

File: newsfragments/1855.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+``dials.image_viewer`` the basis vector colours of multiple crystals now match the prediction colours.
```
Comment (rjgildea) [inline newsfragments/1855.feature:1]:
```suggestion
``dials.image_viewer``: The basis vector colours of multiple crystals now match the prediction colours.
```

File: util/image_viewer/spotfinder_frame.py [hunk 2]
```diff
@@ -1693,7 +1693,11 @@ def map_coords(x, y, p):
                             continue
                         x, y = detector[panel].get_ray_intersection_px(s1)
                         x, y = map_coords(x, y, panel)
-                        vector_data.append((((beam_x, beam_y), (x, y)), vector_dict))
+                        basis_vector_dict = dict(vector_dict)
+                        basis_vector_dict["color"] = self.prediction_colours[i_expt]
+                        vector_data.append(
+                            (((beam_x, beam_y), (x, y)), basis_vector_dict)
+                        )
```
Comment (rjgildea) [inline util/image_viewer/spotfinder_frame.py:1700]: I think this should work, without the need to explicitly create a modified copy of `vector_dict`:
```suggestion
                        vector_data.append(
                            (((beam_x, beam_y), (x, y)), {**vector_dict, "color": self.prediction_colours[i_expt]}),
                        )
```

**Review by rjgildea — APPROVED** [review]
Seems like a sensible change 👍

File: util/image_viewer/spotfinder_frame.py (hunk 2 above)
Comment (dagewa) [reply util/image_viewer/spotfinder_frame.py:1700]: :+1: I'll commit and try it

**Pull request merged**